packages/common/src/authentication/api/api.ts

```diff
@@ -86,7 +86,7 @@ export const getAuthQueries = (sdk: Sdk) => ({
     },
     stores: () => async () => {
       const result = await sdk.me();
-      return result.me?.stores?.nodes;
+      return result?.me?.stores?.nodes;
```

Comment:
there's a few instances of this sort of thing. If the query throws an error and the data returned is undefined, we want to handle that state rather than throwing an unhelpful error at this level.

Reply:
Why is it null? isn't the me return type Promise<MeQuery>?, i.e. never null? or do I miss something?

Reply:
We're using [react-query](https://react-query.tanstack.com/reference/useQuery) for the handling of return objects at the react component level and that is imposing it's own values.

The query method does indeed return a promise, in this case, `Promise<MeQuery>` - but only on success is the data updated. When there is an error, the default `data` value of `undefined` remains unchanged.

Reply:
Ok thanks, so the correct type is `Promise<MeQuery | undefined>`? think we can update the type, can't we?

i.e.
```
export type SdkFunctionWrapper = <T>(action: (requestHeaders?:Record<string, string>) => Promise<T>, operationName: string) => Promise<T | undefined>;
```

Reply:
that export is in the generated file though, and I haven't found an option to modify the result to be potentially undefined. I could do `const result = await sdk.me() as MeQuery | undefined;` and that will pick up the type errors if the undefined isn't handled. I don't like the retyping, but maybe that's better than the alternative